Conversation

EctoOne opened the pull request:
Here is a first version with your background images. The positions might be a little off though. I always do them from what I see because I don't know how to figure them out with a graphic tool. I'm no artist and mostly use paint.net for my graphics or don't use graphics at all.
And grid view is just in an early stage so you have to do some stupid math to position it correctly. I will try if I can find the post about the formula later.
One thing I noticed, was that it seems to be impossible to have that little dot overlapping the selected tile. the zIndex seems to not work with the grid. I tried to make a simple cutout of just the dot and to add it as extra image but then the whole grid disappeared. Actually should probably remove it completely for the same reason as why you can't have game images of screen. For me the dot indicates that always the center tile is selected, and that's just not possible with the current version of the grid.

As for now, I've taken the logos from the carbon theme but I don't know if that's allowed.
Adding the console graphics like you have in your mockups is easy. They just need to have the correct names and should be all in the same subfolder of the _assets folder. You can take a look at the Chicuelo theme for the file structure. https://github.com/chicueloarcade/es-theme-Chicuelo

I've also included just some system folders with the theme.xml which are needed to show the texts on the system view. Also the dates need to be changed for some systems, because I basically just did two (gamegear and nes) and copied them for testing purposes.

EctoOne committed:
1 test

Malixx92 commented:
You rock! Sadly I can't see how it looks on Emulationstation for some reason, the preinstalled themes are working but as soon as I add another one it's blank... I made the changes you asked, no more dot and everything is stored in _assets

EctoOne commented:
What do you mean by it's blank? The theme itself or EmulationStation completely?

Malixx92 commented:
The theme itself is blank but don't worry it must be my EmulationStation I tried with another theme and it does the same, I'm trying to figure it out :/

![es](https://user-images.githubusercontent.com/31268666/44036055-2e2a3ea6-9f11-11e8-9fd9-e73bb45f7428.png)

EctoOne commented:
I'm currently using the latest build from here to create the theme: https://github.com/jrassa/EmulationStation/releases/

EctoOne commented:
Actually my setup is based on this: https://github.com/mattrixk/es-toolkit/releases/download/v1.1/ES_Toolkit_v1.1.7z
Then I merged the latest build into that. Then I run EmulationStation with one of the bat files.

Malixx92 commented:
Thank you, yeah my ES must be outdated, in the meantime I put the theme on my actual Raspberry Pi, it looks good so far! Just a few details to change, like you said the logos are a bit too big in the grey rectangle, but it doesn't have to be pixel perfect if it's too difficult, but it is possible to have them in white and once we are in the gamelist in colors like in my _assets folder?

EctoOne commented:
Yeah, no problem. I just got your files and changed the paths. The Game Gear logo is a little big too high compared to the other ones, if I change to size to fit it in, the others might end up too small. I also noticed that your background images are pngs but have the jpg extension. Since I have access now, I can fix that.
Another thing regarding the grid. You should either include the tiles directly to the background or create a separate file for unselected tiles. Since you're going for that green outline selector it probably would be best to have the tiles directly on the background image or to change the selector image to also have an empty tile as background.

If you want, I can upload a version with the fixed extensions. I'm just adding the theme.xml nothing else, besides renaming both images.

EctoOne commented:
Oh yeah, don't accept the pull request anymore. I just leave it open for the conservation. At least I'm not sure if it will be removed if I close the PR. I have just basic knowledge of how github works. :D

Malixx92 commented:
It's ok I will change the extension and the background tiles, for the Game Gear logo you mean the white one? If yes I must set a max size for every logos in my Photoshop!

I'm new to GitHub too took me 30 minutes to figure out how to create sub folders :p 

Also is it possible to add sounds effect for "Item Selection" and "Launching Game" i've seen it in the NES Mini theme and I think it adds something more to the theme

EctoOne commented:
yes the white one on system view.
As for sounds, I only know of a scrolling sound for the default text gamelist and I can't see anything else in the themes documentation. So, I'm not sure if it will work for the grid. But I will take a look at the NES mini theme to see what's happening there later.

Malixx92 commented:
I made the changes, should be good now!

Malixx92 commented:
Wait I was thinking, maybe it's better if I remove the grey rectangle from the background to put it with the logos so they are aligned?

EctoOne commented:
You mean make the grey rectangle part of the white logos? Yeah that would work and you have a specific canvas where you can put the logos.

Speaking of changing things, I don't know if having the tiles part of the background is a good idea. It makes it harder for me to figure out the correct sizes/positions. It would be great if you could make a single tile that I can use as background for the unselected game and one that has the green overlay. Then I can use the old background and define the grid with the code, not the other way around.

Malixx92 commented:
Ok done! I also put the sounds if you want to try it out and the logo Game Gear with the grey rectangle, I will do all the logos now

EctoOne commented:
Ok, I had to tinker with the images. Had to make icon.png and tile.png the same size and merge icon.png on top of tile.png but it's looking good so far. Maybe you want to do that yourself so that the quality of the images remains.

About the sounds. I don't think that it works for grid and I can't hear a sound when launching a game on the NES mini theme. I will have to look deeper into that, I think I've read something about sounds being broken or something.

Anyway, I'm uploading a version now with the changed images, so please take a look and replace them if you think they suck.
I'm not uploading the system folders, so you won't see the text on system view.

EctoOne commented:
Oh, also don't worry about the dot.png. It's just used for the background color when there's no background image available (e.g. on basic/detailed/video view). It can be removed later.

Malixx92 commented:
Ok I will check this out, also I will change the background to add little arrows on the side like on the 3DS Menu since we can't display all the tiles

Malixx92 commented:
I can't see the grid on my Pi, am I missing something? Is there an update or an option to check?

Malixx92 commented:
https://youtu.be/UB1G2QsH5p8?t=68 (the background music has to be added but the sound effects are in the theme) in nes-mini-master/sounds

Malixx92 commented:
I must really sucks, even after doing exactly what you said:

"Actually my setup is based on this: https://github.com/mattrixk/es-toolkit/releases/download/v1.1/ES_Toolkit_v1.1.7z
Then I merged the latest build into that. Then I run EmulationStation with one of the bat files." 

Still blank. T_T

EctoOne commented:
You have to switch to grid view manually. UI Settings > Gamelist View Style = Grid.

Try downloading this: https://github.com/mattrixk/es-toolkit/archive/v1.2.zip and this https://github.com/jrassa/EmulationStation/releases/download/continuous-master/EmulationStation-Win32.zip
Extract both, then move the content from the second zip into the emulationstation folder from the first zip (not the .emulationstation folder inside the emulationstation folder). That should at least get rid of all the old unused files. Maybe there's some conflict. I tried what I just suggested and it works.

I will check the video later, I'm getting hungry and I need to go to buy some food and stuff.

Malixx92 commented:
Yes thanks it worked! It looks fine, indeed the sound effect with the classic gamelist view but not when I switched to grid view that's sad.

But on the gamelist the tiles are not square anymore, maybe it's me (edit: it is me, on Github your icon.png you did looks fine but not on the theme)

The GBA logo in the gamelist is so little, it's kinda the same problem as before with the Game Gear logo I will figure something out so the logos will be all the same size.

The text in the bubble is really big, could you reduce it a bit?

Is there a way to have the title in only on the top of the bubble and one data under it? I've seen on this video there is at least the release date: https://www.youtube.com/watch?v=qJUqz2lByNo

Thank you and Bon appétit !

EctoOne commented:
Yeah, like I said: Grid View is in an very early stage of development and to be honest, lately there wasn't any development at all. Maybe that's just a thing because it's summer.

As for the tiles, the problem is that the default design is rectangular. I have to find the correct values to make it square and on my screen it looked ok-ish. It also depends on the resolution. Positions and sizes are set by percentage values and not absolute. I designed it with a resolution 3 times as big as you posted (can't check right now but you can get the size from the Screenshots I posted). 

I feel your pain with the different sized logos, that's what always makes it hard to find a solution. In fact that's why I created the Flatline theme, which is pure text based.

Resizing the font and have it in one line only is no problem.
Yet again, I doubt the current version supports any data. I believe the video is from the guy who worked on it but he had to change the code so often that it is just not available yet. I will double check, adding that is quick to do and then I can be certain.

But I'll do that tomorrow, it's getting late.

Malixx92 commented:
Where are you from? It looks like we are in the same timezone

EctoOne commented:
Germany. You?

Malixx92 commented:
France!

Malixx92 commented:
Also the images of the system are a bit too big, I think I have a way to keep the proportions that I want and it would be easier for you to center them! Tell me if it's ok if I do it this way

EctoOne commented:
I've adjusted the font sizes and the system images. Added the release date to the grid (I'm actually surprised that it worked. Glad you kept asking about it.)

I created a simple background image for the text based gamelists, just to get it done and to remove the dot.png. If you ever want to create your own backgrounds for those views, I can adjust the positions of the elements.

I also tried fixing the tile shapes, and I tested it with a resolution of 960x640 and I took a screenshot to see if the tiles are square and they were only like one pixel off. Close enough for now.

I have not looked further into the launching sound yet.

Malixx92 commented:
Nice great job! I think we can forget about the sounds for now because the grid doesn't support it imo

Malixx92 commented:
Maybe I can create separate backgrounds with guidelines on it if that can help you?

EctoOne commented:
that would help. I believe there is a way to figure out positions with photoshop but I have no experience with that. I almost think it would be easier to explain to you how to position things yourself, so that you can do it. It's not that hard to understand. basically an element has a width and height of 100 percent.
And by default the origin point of most elements is the top left corner, but that can be changed with <origin>. So if you want to center an element, the easiest way is to set the origin point to the center first.
<origin>0.5 0.5</origin> does that. it sets the left/top to be at 50% away from those edges. If you then set the position to <position>0.5 0.5</position>, the element will always be centered on the screen.

Malixx92 commented:
Oh you closed the conversation! I uploaded the guidelines but have you tried to put the the new systems images? I think it's better that way and it doesn't require any centering!

EctoOne commented:
I just closed the pull request, that should prevent you from accidentally overwriting the new stuff with the old version. 
I've not looked at the new images yet. The thing is, I wasn't sure if you wanted the system images dead center. I've set them a little bit higher on purpose because I thought dead center would be too close to the bottom box.

Malixx92 commented:
By the way if you need a graphic designer for a future theme of yours I'm here! I have a few started but now that I know the limitations and what is needed I could be faster to product a theme with your help :)

EctoOne commented:
Thanks for the offer, but I barely use my gaming Pi at the moment. But you should look into coding yourself, it is pretty simple. You just need to know which elements exist and then position them. I started by looking at a theme and then changed bits to see what happend. Combining certain settings for different elements or using variables is something that can be done later.
Today it is even easier to code, because now you can simply remove basic elements from a theme without EmulationStation spitting out errors and crashing all the time.
I tried getting into theming for the Pegasus frontend but just reading the manual gave me a headache.

BTW: I updated the positions/sizes according to your guidelines.

Malixx92 commented:
Yeah I should try now that I know it was my EmulationStation that was'nt working, I thought it was my coding.

Malixx92 commented:
It's too bad there is no tutorial video on how to make a theme, I learn better than reading all that stuff! Ok the theme is almost perfect except the tiles, they look really weird to me and the text in the bubble sometimes it centered and sometimes not:

![sans titre](https://user-images.githubusercontent.com/31268666/44098118-9f863a54-9fdf-11e8-819b-09fa95922845.png)

For comparing

![gamelist](https://user-images.githubusercontent.com/31268666/44098195-d13b13b2-9fdf-11e8-8ca7-ae29cd2ef91e.jpg)

EctoOne commented:
Hmm, don't know what's going on with the text. Its origin is in the center and the text is set to be aligned at the center too. So it should expand to both sides equally. The only explanation I can think of is that the text itself wrong. Maybe it has some spaces in the gamelist.xml. The "Released:" text has its origin and alignment on the right side. So the middle is just a bit to the right of the :. If you are running EmulationStation with the --debug parameter, you can press ctrl+t to see the boundaries of some elements. 

For the tiles, I'm out of ideas. It looks fine on my end. Maybe you should ask on the forum for some help with this.

EctoOne commented:
Actually do you want to keep the release date on the grid or should I change it to either developer or publisher?

Malixx92 commented:
Oh well I thought only the release date was possible, so the Dev should be fine! Also I will change the layout of the logos in the gamelist view (I will do the same as the systems images, large png) and the text in the bubble seems little compared to my second screenshot, I should have let you the text in the guidelines:

![guidelines_gamelist](https://user-images.githubusercontent.com/31268666/44100544-98d2ade0-9fe5-11e8-869e-c37d6c9abd50.jpg)

EctoOne commented:
Once again, I've used your guidelines and resized everything to fit right in the boxes you provided. I will take a screen shot in a moment. I also had an idea for the tiles I will test.

Malixx92 commented:
Sorry I must have mixed up the old version of the theme, should I download it directly from the master?

EctoOne commented:
yes

EctoOne commented:
So, I've set the size of the tiles to use exactly a 3:2 ratio, which 240 x 160 is. I've also made the fonts a little bit bigger.
Regarding using full screen images, I won't suggest doing that. It is just a matter of getting the right size. If you create to many big images, you might get VRAM issues. There are two ways to define sizes, <size> and <maxSize>. <size> will stretch/shrink an image to fully fill the defined area. So it is used for something like background images. <maxSize> will only resize until it hits the top/bottom or left/right border, so the aspect ratio will remain of the original image. This is mostly used for logos or other images with transparent background.
From my experience it is best to use a canvas that has the same size for every logo, meaning resize the logo if needed beforehand instead of letting EmulationStation handle it. Like your old gamelist logos have been, the same grey canvas with the logo (although the grey should be removed).

EctoOne commented:
What I can do, is to mark the lines in the theme.xml with the sizes you're unhappy with. Then you can edit them yourself with a texteditor (a good one, but notebook will do) and you can check the results immediately in EmulationStation by refreshing the theme with ctrl+r (if you're running ES with the --debug paramenter)

EctoOne commented:
whoops, forgot to change back to the original backgrounds. also made another small adjustment to the grid.

EctoOne commented:
don't forget that the theme might look weird when you change the images. i noticed that you already changed the gamelist logos but the theme is not prepared for that.

Malixx92 commented:
Yeah I figured the big images will be too much, I will review it a 240 x 160 then! And I will go back to transparent logos

Malixx92 commented:
I can't launch it at 240x160 ES it's closing immediatly

EctoOne commented:
Like I said, I'm using 960x640 to create the theme. It's the same ratio as 240x160 but not too small to actually see things on my PC. You can just copy one of the bat files and edit it.

Malixx92 commented:
I review it at 907x680, well if the tiles are only a problem I can see nvm then, maybe it will work fine with a proper install on a Pi

EctoOne commented:
hmm, here is the content of my bat:
```
set HOME=%~dp0
emulationstation.exe --windowed --debug --resolution 960 640
```

Malixx92 commented:
Haha! I can see the problem now sorry!

Malixx92 commented:
I wanna kill myself, I worked on Photoshop at 907x680... instead of 640

Malixx92 commented:
I don't know why, I took the resolution of the bottom screen of the Nintendo 3DS thinking it was the same as the GBA... I must redo everything

EctoOne commented:
Lol, I thought you were doing that for a portable device that had some weird screen. What display are you actually using? Not that I spend all the time coding for a 3:2 screen when your TV is 16:9.

Malixx92 commented:
It's good 3:2 that the size of the screen of the FreePlay Zero! I am resizing a bit and moving things around to fit everything correctly, so it might have little changes :/

EctoOne commented:
maybe you should provide backgrounds for 16:9 and 4:3 as well. Then I can add options to manually change them (like the Switch theme does that is currently very active on the forum). It would make the theme available on more systems.

Malixx92 commented:
I was aiming only for the portable device, I don't know if it will interest larger screen users since it's designed for a small screen in the first place (big fonts, little informations, etc.)

EctoOne commented:
It's your choice of course, maybe you should ask about it if someone wants a bigger version on the forum. Some people like simple themes though. Well, at least I don't have to figure out the best way to implement that if you don't wanna add it. Meh, i'll probably do it anyway some times just for fun because I'm bored.

Malixx92 commented:
Maybe it would interest people with a GameBoy Zero wich is 3:5 tho! We can do that

EctoOne commented:
Whatever, let's get this going first. Although I've never coded for different resolutions, I have some experience in modular coding, so it should be hard to find a clean way to add this. Especially since the theme has barely any elements that needs to be changed (i believe its just the grid in this case).

Malixx92 commented:
I have many theme if you are bored! Haha I have a Breath of the Wild menu style almost ready to go and a Steam Big Picture :p and a personal one but it requires illustration from me!

EctoOne commented:
It depends, I'm mostly up for a challenge to break the limitations of theme creation. But sometimes I ran into a wall and ask for things to be added to EmulationStation but it's hard to get them. I'm not complaining, some things have been added because I asked for them, but some simple things (at least I think they are simple) are not. And that is even weirder when some elements already support them. For example: rotation. There is a way to freely rotate text and images but not textlists. And I really needed a tilted gamelist when I was trying to make a comic theme.

Malixx92 commented:
Yeah I thought the same, maybe you should watch the Recalbox, it's similar to RetroPie but its made by French people and they are ahead on some features on their EmulationStation(?) look a this:

https://youtu.be/iyCPaF95KYc?t=158

The gamelist is not tilted but the carousel is and I've never seen that on Retropie forums. They had the collections, favorites etc. way before Retropie.

Also I just added new backgrounds and new guidelines that fit the resolution, some element have moved even if it doesn't seems like it.

EctoOne commented:
Oh doing a wheel as carousel is also possible on retropie. It almost seems that recallbox is adapting everything retropie has, add some extra features (like being able to change the carousel from the menu). It's so weird that the versions are so different, and that the people are not working together. I prefer retropie because it is a little more advanced on the systems side but EmulationStation on recallbox has some nice features.

Malixx92 commented:
For exemple they have a background music that comes with the themes, never seen that on retropie except by doing some weird stuff, and it's not very clean because when you launch a game you can see the code lines of the music pop on the screen!

Anyway I will check the theme now that the images are correct.

EctoOne commented:
I'm more interested by the fact that you can change theme options directly from the GUI so that users don't have to edit the files manually. And that you can theme the GUI. I hate the white version from Retropie.

I've not changed anything yet in case you uploaded new files.

Malixx92 commented:
Yeah I figured, the elements are just a bit off now, but the tiles are still weird to me (less than my old resolution), can you screenshot what you see please?

EctoOne commented:
Here. I just noticed that I have to fix the padding of the tiles a little bit, made it to small yesterday. Meaning the images are to close to the edges in my opinion.

![unbenannt](https://user-images.githubusercontent.com/30272930/44147771-c29825e6-a095-11e8-86ea-36fe221b03cb.png)

Malixx92 commented:
Yes I updated the guidelines, you can see the padding there, you have also the old background so its stretched but I updated that aswell, but the tiles are very small on your screenshot compared to mine

Malixx92 commented:
But we are almost there!

EctoOne commented:
yeah, im trying to making them bigger right now.

EctoOne commented:
seems like you messed up the tile and icon sizes again. they need to be the same size. this is getting annoying

EctoOne commented:
also the green overlay looks weird because there's no way to avoid it from being stretched. at least no way i know of.

Malixx92 commented:
Ok I changed it, also I removed the grey background on the gamelist logos

EctoOne commented:
So, here it is with the guidelines as backgrounds.
![unbenannt](https://user-images.githubusercontent.com/30272930/44150202-f59f8f16-a09e-11e8-82b1-3cd759d01c12.png)

Malixx92 commented:
Nice thank you! This seems all good

Malixx92 commented:
The tiles seems a little bit stretched but it's close enough, way better than before!

EctoOne commented:
the only other thing i noticed, is that the system images are not in your guideline box when i use the full screen images you provided. they are to low, i can fix that with the code but i would suggest to go back to the small versions you had before. i figured out that i made a mistake before, that's why they looked compressed.

Malixx92 commented:
Ok I will upload the croped images of the systems

Malixx92 commented:
All set up! Some are not finished but I can change that later, they are all the same size (they don't fit the guideline but align them at the bottom of the red square and it should work)

EctoOne commented:
Done, the origin is at the bottom of the box and the max size is set to fit right in the box as well.

Malixx92 commented:
It's almost perfect, I'm just not satisfied yet with the systems emplacements so I updated the guidelines for that, the red square should fit perfectly the png. The tiles seems perfect now!

EctoOne commented:
What I did to find the size now was that I filled the background one of the current system images with a solid color. This way I could see the edges of the image. I started with a small image and made it bigger until it reached the borders. I actually don't understand what you want now.

EctoOne commented:
Oh man, i just took a look at the new guideline image. I have to reposition everything -.-

EctoOne commented:
ok here is the problem. the background image gets stretched to fit my set screen resolution. your image is 907x640, my resolution is 960x640. That's why I'm setting the element sizes/position on a stretched background. Which obviously will be off.
Solutions:
* I can change my resolution to your image size (why is it even 907?)
* I can align the background image to the center and to not stretch, but then if the resolution is bigger there will be borders. Or if the resolution is smaller, the image will be cut off.

So you have to tell me what resolution you really want.

Malixx92 commented:
Wait I'm looking at other themes to be sure what resolution to use, maybe it shouldn't be so big. Some are using 4:3 and others 640x480.

Malixx92 commented:
Don't touch anything for now I found some post talking about it

Malixx92 commented:
I opted for the 640x480, like that nothing can go wrong, it was used on the previous theme for the Freeplay. But it seems that I have to rescale everything I already put the guidelines and backgrounds for you, but I think I have to shrink down the rest of the images (logos and systems) I don't think you can do that on the coding side.

Malixx92 commented:
I scaled down every images, so don't worry about that!

EctoOne commented:
since the tiles are supposed to be squared, why are the images not square? even if i could easily get the values correct right away for square tiles, i still have to adapt to your sizes.

EctoOne commented:
done

Malixx92 commented:
The tiles have a shadow effect so the images are not totally a square, but I can change that by expanding the transparency (I didn't think it could be an issue)

Malixx92 commented:
It looks perfect to me, thank you very much!

Malixx92 commented:
Sometimes I update images but GitHub keeps the old one... I keep reuploading it and it won't change!

EctoOne commented:
I don't know why, maybe it's just a cache thing.
The non square image are not particularly an issue, just something that seemed weird to me.

Malixx92 commented:
I've noticed small details:

- The publisher names are bigger than the game titles
- The gamelist_logos are a bit too much to the right (like 1 or 2 pixels to the left should be good)
- The system infos are missing (use FOT-RodinBokutohPro-B.otf for them)

EctoOne commented:
* Hmm, you're right.
* The origin and position is set to 0.5, so they should be centered. It's either a bug or your images are not centered.
* I will create one or two system infos, you can then copy/paste the folders yourself and edit the text files. I'm not going through every system to find it's release date, sorry.

But I'm pretty busy today and will do it tonight.

Malixx92 commented:
Of course I will do it, I thought there was a "scraper" for the systems, but no problem.

EctoOne commented:
* Fixed the fonts
* tried to fix the logo positions (i added an outline to one of the logos as you can see here, it fits perfectly on my screen).
![unbenannt](https://user-images.githubusercontent.com/30272930/44236786-7f7bc780-a1ae-11e8-8408-94c672a0b29b.png)
* added a gamegear folder with the theme.xml. to add a system: copy the folder, rename it according to the correct system name, edit the new theme.xml, should be pretty clear what to do. Replace the year and the developer name.

Malixx92 commented:
Thank you! I will look at it, should be simple enough. The gamelist logos are good also the system logo is a bit off ?

EctoOne commented:
Sorry about the late answer, I was very busy this week. By system logo you mean the one that's next to the text right? It might be, because the guideline for that has just an logo, not a pink box. So I might have overlooked it. I will check it later.

Malixx92 commented:
No problem! Well yeah, anyway it's close enough so don't worry, thank you! Now all the work is on my side (images, logos...)

Malixx92 commented:
Hello EctoOne,

I'm sorry to bother you again, but I'm working on another theme and I got the hang of it but I can't find any tutorials about the grid view and I'm a bit lost...

I tried to do it all alone but I'm asking for your help, I can't figure it out on my own. :(

It would be amazing if you would help me just for this part of the theme.

Have a nice day!

Malixx92 commented:
Here is the link to the theme if you have time to check it out:

https://mega.nz/#F!TKRkAY6S!bxMYsNIM6RzcdQllCNRhIA

The theme is almost done but I have trouble with the carousel and the grid as you can see!

EctoOne commented:
Sorry, but I lost interest in working on this stuff a while ago. And I'm not in the mood right now to get back into it. Try asking on the official Retropie forum.
